build_tools/i18n/crowdin.py

```diff
@@ -326,8 +326,17 @@ def _locale_data_from_csv(file_data):
     for row in file_data:
         if len(row.keys()) == 0:
             return json
-        # First index is Identifier, Third index is the translation
-        json[row["Identifier"]] = row["Translation"]
+        # First index is Identifier
+        # Third index is the translation, but might have a key 'None'
+        json[row["Identifier"]] = row.get("Translation") or row.get(None)
+
+        # hack - sometimes it's a list?
+        if isinstance(json[row["Identifier"]], list):
+            json[row["Identifier"]] = json[row["Identifier"]][0]
+
+        # hack - sometimes it's None?
+        if not json[row["Identifier"]]:
+            json[row["Identifier"]] = ""
```

Comment:
This is weird - but the row ought to be a dict where the keys are gathered by the header row of the CSV that was parsed on line ~ 314 `csv_data = list(row for row in csv.DictReader(f))` - if a key is None, then the CSV likely is missing a header.

Reply:
Also I see no reason why it wouldn't be a list.

Did you preapprove all of the translations on whatever branch you were using for testing? Missing translations is my best guess but I'm not sure. I've never encountered this while testing with a fully translated branch though.

Reply:
And - if that's the case - it might be weird to have incomplete translations as well.

> So to be clear, I should pull the latest from crowdin and add them to this PR?

Would it have been easier (albeit a bit tedious) to write a script to rename all of the files in the locale directory to avoid pulling incomplete translations into the repo for now (assuming that's the case).

Reply:
reverted the change, re-ran everything, and now .. it works?